sharding/networksharding/peerShardMapper.go

```diff
@@ -274,16 +282,17 @@ func (psm *PeerShardMapper) UpdatePeerIDInfo(pid core.PeerID, pk []byte, shardID
 		return
 	}
 	psm.updatePublicKeyShardId(pk, shardID)
-	psm.updatePeerIdShardId(pid, shardID)
+	psm.UpdatePeerIdShardId(pid, shardID)
 	psm.preferredPeersHolder.Put(pk, pid, shardID)
 }
 
 func (psm *PeerShardMapper) updatePublicKeyShardId(pk []byte, shardId uint32) {
-	psm.fallbackPkShardCache.HasOrAdd(pk, shardId, uint32Size)
+	psm.fallbackPkShardCache.Put(pk, shardId, uint32Size)
 }
 
-func (psm *PeerShardMapper) updatePeerIdShardId(pid core.PeerID, shardId uint32) {
-	psm.fallbackPidShardCache.HasOrAdd([]byte(pid), shardId, uint32Size)
+// UpdatePeerIdShardId adds the peer ID and shard ID into fallback cache in case it does not exists
+func (psm *PeerShardMapper) UpdatePeerIdShardId(pid core.PeerID, shardId uint32) {
+	psm.fallbackPidShardCache.Put([]byte(pid), shardId, uint32Size)
```

Comment:
maybe check if already exists? since the function is called update

Reply:
as discussed, change the method name to `PutPeerIdShardId`